lib/client/doc.js

```diff
@@ -186,9 +186,9 @@ Doc.prototype.ingestSnapshot = function(snapshot, callback) {
   if (this.version > snapshot.v) return callback && callback();
 
   this.version = snapshot.v;
-  this.data = snapshot.data;
   var type = (snapshot.type === undefined) ? types.defaultType : snapshot.type;
   this._setType(type);
+  this.data = (this.type && this.type.serialize) ? this.type.deserialize(snapshot.data) : snapshot.data
```

Comment:
i'm concerned that setting this.data after the call to this._setType() is problematic since _setType() sets this.data to null if the type is null, but we need access to the type in order to deserialize the snapshot.data.  Thoughts?